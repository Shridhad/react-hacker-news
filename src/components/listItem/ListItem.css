.list-item {
    padding: 0.5rem;
    margin-left: 1rem;
    white-space: normal;
    border-bottom: 1px solid hsla(214, 53%, 23%, 0.16);  
}

.list-item .item-title a {
    color: #333;
    text-decoration: none;
    font-weight: 500;
}

.list-item .item-details {
    color: #666;
}

.list-item .item-details .item-meta {
    display: inline-block;
    padding: 0 0.3rem 0 0;
}

.list-item .item-meta a {
    color: #666;
    text-transform: lowercase;    
    text-decoration: none;
}

.list-item .item-meta a:hover {
    text-decoration: underline;
}

.list-item .item-details .time-ago:after {
    content: "|";
    padding-left: 2px;        
}