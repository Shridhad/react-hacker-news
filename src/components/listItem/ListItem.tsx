import React from 'react';
import StoryItem from '../../models/StoryItem';

import './ListItem.css';

export default class ListItem extends React.Component<StoryItem, {}> {
    render() {
        return (
            <div className="list-item">
                <div className="item-title">
                    <a href={this.props.url} target="_blank">{this.props.title}</a>
                </div>
                <div className="item-details">
                    <span className="item-meta points">{this.props.points} Points</span>
                    <span className="item-meta user">by <a target="_blank" href={`https://news.ycombinator.com/user?id=${this.props.user}`}>{this.props.user}</a>
                    </span>
                    <span className="item-meta time-ago">
                        <a href={`/item/${this.props.id}`}>{this.props.time_ago}</a>
                    </span>
                    <span className="item-meta comments-count">
                        <a href={`/item/${this.props.id}`}>{this.props.comments_count} Comments</a>
                    </span>
                </div>
            </div>
        )
    }
}