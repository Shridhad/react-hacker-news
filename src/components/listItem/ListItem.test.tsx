import React from 'react';
import ReactDOM from 'react-dom';
import ListItem from './listItem';

describe("<HomeView />", () => {
    it("renders HomeView with stories", () => {
        const story = {
            "id": 19780387,
            "title": "Topics in Advanced Data Structures [pdf]",
            "points": 131,
            "user": "htiek",
            "time": 1556558376,
            "time_ago": "2 hours ago",
            "comments_count": 16,
            "type": "link",
            "url": "http://web.stanford.edu/class/cs166/handouts/100%20Suggested%20Final%20Project%20Topics.pdf",
            "domain": "web.stanford.edu"
        };

        const div = document.createElement('div');
        ReactDOM.render(<ListItem {...story} />, div);
        
        expect(div).toMatchSnapshot();
        ReactDOM.unmountComponentAtNode(div);
    });
});
