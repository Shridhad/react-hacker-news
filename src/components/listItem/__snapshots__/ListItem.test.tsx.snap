// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView /> renders HomeView with stories 1`] = `
<div>
  <div
    class="list-item"
  >
    <div
      class="item-title"
    >
      <a
        href="http://web.stanford.edu/class/cs166/handouts/100%20Suggested%20Final%20Project%20Topics.pdf"
        target="_blank"
      >
        Topics in Advanced Data Structures [pdf]
      </a>
    </div>
    <div
      class="item-details"
    >
      <span
        class="item-meta points"
      >
        131
         Points
      </span>
      <span
        class="item-meta user"
      >
        by 
        <a
          href="https://news.ycombinator.com/user?id=htiek"
          target="_blank"
        >
          htiek
        </a>
      </span>
      <span
        class="item-meta time-ago"
      >
        <a
          href="/item/19780387"
        >
          2 hours ago
        </a>
      </span>
      <span
        class="item-meta comments-count"
      >
        <a
          href="/item/19780387"
        >
          16
           Comments
        </a>
      </span>
    </div>
  </div>
</div>
`;
